frontend/lib/generate-image-markup.ts

```diff
@@ -18,7 +18,8 @@ export function markdown(
   link?: string,
   title?: string
 ): string {
-  const withoutLink = `![${title || ''}](${badgeUrl})`
+  const withoutSpaces = badgeUrl.replace(/\s/g, '%20')
+  const withoutLink = `![${title || ''}](${withoutSpaces})`
```

Comment:
Hi! Thanks for this and for looking into a fix!

I think the problem is not here, but in the badge URL, which should be a valid URL before it's passed to this function. Fixing it here would only fix it for Markdown though the problem exists for each of these formats since the URL is what's wrong.

Reply:
@paulmelnikow Ok, that makes sense. I'm not too familiar with the code base. Would this be a better spot to implement a fix? https://github.com/badges/shields/blob/b4579adc44329e33c7491a78f14a53a7b30ce34c/frontend/components/customizer/customizer.tsx#L51

Reply:
I think it needs to be here:

https://github.com/badges/shields/blob/b4579adc44329e33c7491a78f14a53a7b30ce34c/frontend/components/customizer/path-builder.tsx#L81-L108